torch/_inductor/fx_passes/quantization.py

```diff
@@ -1229,10 +1306,43 @@ def _inner(match):
     return _inner
 
 
-def _register_dequant_promotion_pass(pattern, pass_number, dtype=torch.float32):
+def _is_valid_woq_mm_dequant_promotion_pattern(
+    end_node_target_candidates, dtype=torch.bfloat16
+):
+    def _inner(match):
+        assert dtype == torch.bfloat16
+        dequant_pattern_end_node = match.output_node()
+        if dequant_pattern_end_node.target not in end_node_target_candidates:
+            return False
+
+        unsqueeze_node = dequant_pattern_end_node
+        convert_node = unsqueeze_node.args[0]
+        reshape_node = convert_node.args[0]
+        if (
+            unsqueeze_node.target is aten.unsqueeze.default
+            and convert_node.target is prims.convert_element_type.default
+            and convert_node.args[1] == torch.float32
+            and reshape_node.target is aten.reshape.default
+            and len(list(dequant_pattern_end_node.users)) > 1
+        ):
+            # If dequant pattern has more than 1 users, then do dequant promoted
+            return True
+        return False
+
+    return _inner
+
+
+def _register_dequant_promotion_pass(
+    pattern,
+    pass_number,
+    extra_check_fn,
+    find_first_node_fn,
+    end_node_target_candidates,
+    dtype=torch.float32,
```

Comment:
Can you add function doc to explain what the function does and what these arguments mean?

Reply:
Thanks and added.